pyproject.toml

```
@@ -198,15 +199,19 @@ omit = [
 
 [tool.pytest.ini_options]
 doctest_optionflags = 'NUMBER ELLIPSIS'
+
+# Keep spaces between filter warnings since order matters (see https://docs.pytest.org/en/stable/how-to/capture-warnings.html#controlling-warnings)
 filterwarnings = [
-  'error::pyvista.PyVistaDeprecationWarning',
-  'ignore:.*Given trait value dtype "float64":UserWarning', # bogus numpy ABI warning (see numpy/#432)
-  'ignore:.*The NumPy module was reloaded*:UserWarning',    # bogus numpy ABI warning (see numpy/#432)
-  'ignore:.*numpy.dtype size changed.*:RuntimeWarning',     # bogus numpy ABI warning (see numpy/#432)
-  'ignore:.*numpy.ufunc size changed.*:RuntimeWarning',     # bogus numpy ABI warning (see numpy/#432)
-  'ignore::DeprecationWarning',
-  'ignore::FutureWarning',
-  'ignore::PendingDeprecationWarning',
+  'error',
```

This line broke debugging for me in PyCharm. If I add breakpoints, the execution isn't stopped anymore. If I comment out this line though, debugging with breakpoints works again.

---

Sorry to hear that.

I am not a PyCharm user so I could not see the issue. On my side, VSCode works fine with these settings.

Does it happen for any python code or specifically with the tests ?

---

I created a new Codespace and can reproduce the issue there. To reproduce:

1. Create a new Codespace from main
2. Add a breakpoint to a test file, e.g. add one at the top of the file at one of the import statements.
3. Start debugging any test in the test file. Execution is expected to stop at the breakpoint but it doesn't. The test will still run to completion though and pass.

Workaround to fix this:

1. Comment out or remove `'error'` from `filterwarnings`
2. Debug the tests again. Execution should stop at the breakpoint (as expected).

This is a very peculiar issue, maybe it's a `pytest` bug?

---

I was able to reproduce the behavior on VScode. I guess a matplotlib warning is emitted by the python debugger. 

Try add that in the `pyproject.toml`
```suggestion
'ignore:The .*interactive_bk.* attribute was deprecated in Matplotlib 3\.9.*:matplotlib.MatplotlibDeprecationWarning',
```

It fixed breakpoints on my side